app/src/main/java/org/dosomething/letsdothis/ui/CampaignDetailsActivity.java

```diff
@@ -60,9 +100,77 @@ protected void onDestroy()
         super.onDestroy();
     }
 
-    public static Intent getLaunchIntent(Context context, int campaignId)
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data)
     {
-        return new Intent(context, CampaignDetailsActivity.class).putExtra(EXTRA_CAMPAIGN_ID, campaignId);
+        if(resultCode == RESULT_OK)
+        {
+            if(requestCode == SELECT_PICTURE)
+            {
+                final boolean isCamera;
+                if(data == null)
+                {
+                    isCamera = true;
+                }
+                else
+                {
+                    final String action = data.getAction();
+                    if(action == null)
+                    {
+                        isCamera = false;
+                    }
+                    else
+                    {
+                        isCamera = action.equals(android.provider.MediaStore.ACTION_IMAGE_CAPTURE);
+                    }
+                }
+
+                Uri selectedImageUri;
+                if(isCamera)
+                {
+                    selectedImageUri = imageUri;
+                }
+                else
+                {
+                    selectedImageUri = data == null
+                            ? null
+                            : data.getData();
+                }
+                if(BuildConfig.DEBUG)
+                {
+                    Log.d("asdf-----------", selectedImageUri.toString());
+                }
+                ImageView viewById = (ImageView) findViewById(R.id.test);
+                Picasso.with(this).load(selectedImageUri).into(viewById);
+            }
+        }
+    }
+
+
+    public void choosePicture()
```

Comment:
FIXME: I get "Android System" as an option to choose photos. I don't think that should happen (?)